backend/src/endpoints/user/google.ts

```diff
@@ -0,0 +1,77 @@
+import { Request, Response } from "express";
+import {
+    handleGoogleLoginRequest,
+    verifyGoogleToken,
+    updateAlgoToken,
+    verifyAlgoToken,
+} from "../../services/authservice";
+
+export const googleLogin = async (req: Request, res: Response) => {
+    // get google token, verify it, create AlgoToken for user
+    const token = req.params.token;
+    const nickname = req.body.nickname;
+
+    const payload = await verifyGoogleToken(token);
+
+    if (!payload) {
+        res.status(400).send("Invalid Google token");
+        return;
+    }
+
+    const AlgoToken = await handleGoogleLoginRequest(payload, nickname);
+
+    if (!AlgoToken) {
+        res.status(500).send();
+        return;
+    }
+
+    res.cookie("AlgoToken", AlgoToken, { httpOnly: true }); // secure: true for https
+    res.status(200).json({ logedIn: true });
+};
+
+export const googleLogout = async (req: Request, res: Response) => {
```

Comment:
Nie wiem czy tworzenie funkcji logout dla każdego sposobu logowania jest najlepsze. Myślę, że jedna by wystarczyła. To logika dla AlgoTokena, nie GoogleTokena.